scylla-cdc-replicator/src/replication_tests.rs

```diff
@@ -145,6 +147,10 @@ mod tests {
                 o, r
             ),
             FailureReason::RowNotMatching(n) => eprintln!("Row {} is not equal in both tables.", n),
+            FailureReason::TimestampsNotMatching(row, column) => eprintln!(
+                "Timestamps were not equal for column {} in row {}.",
```

Comment:
The code below seems to assign numbers to rows starting from 1, IIUC here you start from 0. (should print `row + 1`)

Reply:
Do you agree with this comment? If yes: it's still printing `row`.

Reply:
I agree with this comment, but I don't see a problem. It shows a correct row number in the test `test_checking_timestamps`, where failing for not matching timestamps is tested. 
`FailureReason::TimestampsNotMatching` in function `compare_timestamps` is constructed as `FailureReason::TimestampsNotMatching(i + 1, col_name.to_string())`, so this `row` here is in fact `row+1`.

Reply:
I think the `TimestampsNotMatching` should keep a zero-based row index. In Rust, indexing is zero-based, so unless it is documented, the row index being one-based might be confusing. It's better to stick to the convention.

Another problem is that, given that most indexes are zero-based, usually you will have to remember to add `1` every time you construct `TimestampsNotMatching` and it's easy to forget that. If you switch to zero-based index there will be a smaller chance of error and adding `1` will only be a concern of the logic printing the error report.

Reply:
> the row index being one-based might be confusing.

I don't think that's the case since rows of the faulty column are enumerated after the main error message, so whoever will look into these errors should see that indexing is one-based. Example of an error message:
```
Replication test for table COMPARE_TIME failed.
Failure reason: Timestamps were not equal for column v2 in row 2.
ORIGINAL TABLE:
Row 1: [Some(BigInt(1645814895337319))]
Row 2: [Some(BigInt(1645814895378103))]
REPLICATED TABLE:
Row 1: [Some(BigInt(1645814895337319))]
Row 2: [Some(BigInt(1645814895338050))]
```

Reply:
I meant that using a one-based index _in the code_ is confusing; you should store a zero-based index inside the `TimestampsNotMatching` variant. The conversion to one-based index should be made just before printing.

Reply:
Okay, I have changed this along your suggestion.